src/main/java/org/ibp/api/java/impl/middleware/inventory/manager/TransactionServiceImpl.java

```diff
@@ -220,4 +223,44 @@ public void saveDeposits(final LotDepositRequestDto lotDepositRequestDto, final
 			lock.unlock();
 		}
 	}
+
+	@Override
+	public void cancelPendingTransactions(final SearchCompositeDto searchCompositeDto) {
+		final BindingResult errors = new MapBindingResult(new HashMap<String, String>(), TransactionService.class.getName());
+
+		try {
+			lock.lock();
+
+			//Validate that searchId or list of lots are provided
+			if (searchCompositeDto.getSearchRequestId() == null && (searchCompositeDto.getItemIds() == null || searchCompositeDto
+				.getItemIds().isEmpty()) ||
+				(searchCompositeDto.getSearchRequestId() != null && searchCompositeDto.getItemIds() != null)) {
+				errors.reject("transaction.selection.invalid", "");
+				throw new ApiRequestValidationException(errors.getAllErrors());
+			}
+
+			TransactionsSearchDto transactionsSearchDto;
+			if (searchCompositeDto.getSearchRequestId() != null) {
+				transactionsSearchDto =
+					(TransactionsSearchDto) this.searchRequestService
+						.getSearchRequest(searchCompositeDto.getSearchRequestId(), TransactionsSearchDto.class);
+			} else {
+				transactionsSearchDto = new TransactionsSearchDto();
+				transactionsSearchDto.setTransactionIds(new ArrayList<>(searchCompositeDto.getItemIds()));
+			}
+
+			final List<TransactionDto> transactionDtos = this.transactionService.searchTransactions(transactionsSearchDto, null);
+			final Set<ExtendedLotDto> lotDtos = transactionDtos.stream().map(TransactionDto::getLot).collect(
+				Collectors.toSet());
+
+			transactionInputValidator.validateAllProvidedTransactionsExists(transactionDtos, searchCompositeDto.getItemIds());
+			transactionInputValidator.validateEmptyList(transactionDtos);
```

Comment:
@cuenyad  validateEmptyList should go before validateAllProvidedTransactionsExists